torch/_higher_order_ops/while_loop.py

```diff
@@ -246,16 +321,41 @@ def _trace_while_loop(
 
 
 def check_outputs_carry_consistency(
-    outs: List[torch.Tensor], carries: List[torch.Tensor]
+    outs: List[Union[torch.Tensor, torch.SymInt, int]],
+    carries: List[Union[torch.Tensor, torch.SymInt, int]],
 ) -> None:
-    all_diffs_in_meta = []
-    for out, cry in zip(outs, carries):
-        if diff := diff_tensor_meta(
-            _extract_tensor_metadata(cry), _extract_tensor_metadata(out)
-        ):
-            all_diffs_in_meta.append(",".join(diff))
-    if all_diffs_in_meta:
-        diff_str = "\n".join(all_diffs_in_meta)
+    def diff_meta_pairs(
+        lhs_list: List[Union[torch.Tensor, torch.SymInt, int]],
+        rhs_list: List[Union[torch.Tensor, torch.SymInt, int]],
+    ) -> List[str]:
+        def diff_meta(
+            lhs: Union[torch.Tensor, torch.SymInt, int],
+            rhs: Union[torch.Tensor, torch.SymInt, int],
+        ) -> str:
+            if isinstance(lhs, torch.Tensor) and isinstance(rhs, torch.Tensor):
+                # We have vmap x cond tests and querying is_contiguous inside of vmap for
+                # memory_format other than torch.contiguous_format is not yet implemented.
+                # And it seems the remaining metas are good enough for now.
```

Comment:
I don't understand what this comment means

Reply:
This is like a corner test case that we're make_fx over vmap x cond, when it reaches here, if we're try to set include_contiguity=True, it will error out saying  `querying is_contiguous inside of vmap for memory_format other than torch.contiguous_format is not yet implemented`. So we have to set include_contiguity=False.